js/movie_script.js

```diff
@@ -0,0 +1,177 @@
+import {fillDropdown, postOrPutObjectAsJson, restDelete, ageLimits, movieCategories} from "./module.js";
+
+const category = document.getElementById('category')
+const ageLimit = document.getElementById('age_limit')
+
+fillDropdown(category, movieCategories, 'category');
+fillDropdown(ageLimit, ageLimits, 'age limit');
+
+
+/*----------------------------------------------FORM---------------------------------------------------*/
+document.addEventListener('DOMContentLoaded', createFormEventListener);
+
+let movie_form;
+
+function createFormEventListener() {
+    movie_form = document.getElementById('movie_form');
+    movie_form.addEventListener('submit', handleSubmitForm);
+}
+
+async function handleSubmitForm(event) {
+    event.preventDefault();
+    const form = event.currentTarget;
+    const url = form.action;
+
+    try {
+        const formData = new FormData(form);
+        await postAndPutFormDataAsJson(url, formData);
+    } catch(error) {
+        alert(error.message); //make error message appear above the 'create form' instead
+    }
+}
+
+async function postAndPutFormDataAsJson(url, formData) {
+    const plainFormData = Object.fromEntries(formData.entries());
+    console.log(typeof plainFormData.movieID);
+    let response;
+    if(plainFormData.movieID !== '') {
+        response = await postOrPutObjectAsJson(url, plainFormData, 'PUT');
+        if(response.ok) {
+            alert('MOVIE UPDATED'); //make error message appear above the 'create form' instead
+        }
+    } else {
+        response = await postOrPutObjectAsJson(url, plainFormData, 'POST');
+        if(response.ok) {
+            alert('MOVIE CREATED'); //make error message appear above the 'create form' instead
+        }
+    }
+
+}
+
+/*--------------------------------------------FETCH DATA-------------------------------------------------*/
+
+const movie_url = "http://localhost:8080/movies";
```

Comment:
Change to azure backend checkpoint.
"https://kinoxp-back.azurewebsites.net/movies"